Fix issue related to enrollment URL's and refresh screen on successful enrollment

shahidtamboli opened the pull request:
There were a few URL changes in the enrollment feature. We have updated it according to the new URL's
The page was not getting auto refreshed on successful enrollment. This has been fixed.

@rohan-dhamal-clarice @aleffert - Please review

rohan-dhamal-clarice commented on android/source/VideoLocker/androidTest/java/org/edx/mobile/test/UrlTests.java:
```diff
@@ -24,12 +25,25 @@ public void testReadUrlPaths() throws Exception {
+        print("Path Id - "+pathId);
+    }
+
+    /*public void testCourseInfoPaths() throws Exception {
```
Please remove this dead code.

rohan-dhamal-clarice commented on android/source/VideoLocker/androidTest/java/org/edx/mobile/test/UrlTests.java:
```diff
@@ -24,12 +25,25 @@ public void testReadUrlPaths() throws Exception {
+
+        assertNotNull(pathId);
+        assertEquals("cosmology-anux-anu-astro4x", pathId);
+        print("Path Id - "+pathId);
```
<code>print()</code> not required. Please remove.

rohan-dhamal-clarice commented on android/source/VideoLocker/src/org/edx/mobile/view/dialog/EnrollmentFailureDialogFragment.java:
```diff
@@ -50,7 +50,7 @@ public View onCreateView(LayoutInflater inflater, ViewGroup container,
         // Watch for button clicks.
         Button positiveBtn = (Button) v.findViewById(R.id.positiveButton);
-        positiveBtn.setText(getArguments().getString("dialog_yes_btn"));
+        positiveBtn.setText(getArguments().getString("dialog_yes_btn").toUpperCase());
```
Instead, use <code>android:textAllCaps</code> property for the <code>TextView</code> in XML.  This is good, if this dialog class is supposed to show some other contents where buttons do not have text in ALL Caps.

rohan-dhamal-clarice commented:
Otherwise, looks good.

shahidtamboli commented:
@rohan-dhamal-clarice - I have updated the code as per comments. I have also removed the code where "You are currently not enrolled" message shows up. That message was not required to be displayed as suggested by Lou in one of the UI reviews.

rohan-dhamal-clarice commented:
:+1:

aleffert commented on android/source/VideoLocker/src/org/edx/mobile/base/BaseFragmentActivity.java:
```diff
@@ -764,6 +763,13 @@ private boolean showErrorMessage(String header, String message) {
     /**
+     * This method should be implemented by {@link org.edx.mobile.view.MyCoursesListActivity}.
+     */
+    protected void reloadMyCoursesData() {
```
Having logic specific to *one* activity in the class *all* activities subclass from seems weird to me. Maybe we should better configure how we handle these messages.

aleffert commented on android/source/VideoLocker/src/org/edx/mobile/view/BaseTabActivity.java:
```diff
@@ -62,6 +63,15 @@ private void setUpTabs() {
     }
 
+    protected Fragment getFragmentByTag(String tag) {
+        if (fragmentManager != null) {
```
How does the fragment manager end up null? Because we're on an old version of android? Maybe worth adding a comment.

aleffert commented on android/source/VideoLocker/src/org/edx/mobile/view/custom/URLInterceptorWebViewClient.java:
```diff
@@ -27,13 +27,13 @@
-    private static final String PARAM_PATH_ID           = "path_id";
+    private static final String URL_TYPE_ENROLL         = "edxapp://enroll";
+    private static final String URL_TYPE_COURSE_INFO    = "edxapp://course_info";
+    //public static final String URL_TYPE_COURSE_INFO    = "edxapp://view_course/course_path=course/";
```
We should remove this line instead of commenting it out?
